src/mod_shared_roster_ldap.erl

```diff
@@ -105,34 +105,48 @@ stop(Host) ->
-                 <<"ldap_memberattr_format_re">> => #option{type = string},
-                 <<"ldap_auth_check">> => #option{type = boolean},
-                 <<"ldap_user_cache_validity">> => #option{type = integer,
-                                                           validate = positive},
-                 <<"ldap_group_cache_validity">> => #option{type = integer,
-                                                            validate = positive},
-                 <<"ldap_user_cache_size">> => #option{type = integer,
-                                                       validate = positive},
-                 <<"ldap_group_cache_size">> => #option{type = integer,
-                                                        validate = positive},
-                 <<"ldap_rfilter">> => #option{type = string},
-                 <<"ldap_gfilter">> => #option{type = string},
-                 <<"ldap_ufilter">> => #option{type = string},
-                 <<"ldap_filter">> => #option{type = string}
-                }
-      }.
+    CommonLDAPSpec = mongoose_ldap_config:spec(),
+    Items = #{<<"groupattr">> => #option{type = binary},
+              <<"groupdesc">> => #option{type = binary},
+              <<"userdesc">> => #option{type = binary},
+              <<"useruid">> => #option{type = binary},
+              <<"memberattr">> => #option{type = binary},
+              <<"memberattr_format">> => #option{type = binary},
+              <<"memberattr_format_re">> => #option{type = binary},
+              <<"auth_check">> => #option{type = boolean},
+              <<"user_cache_validity">> => #option{type = integer,
+                                                   validate = positive},
+              <<"group_cache_validity">> => #option{type = integer,
+                                                    validate = positive},
+              <<"user_cache_size">> => #option{type = integer,
+                                               validate = positive},
+              <<"group_cache_size">> => #option{type = integer,
+                                                validate = positive},
+              <<"rfilter">> => #option{type = binary},
+              <<"gfilter">> => #option{type = binary},
+              <<"ufilter">> => #option{type = binary}
+    },
+    Defaults = #{<<"groupattr">> => <<"cn">>,
+                 <<"userdesc">> => <<"cn">>,
+                 <<"useruid">> => <<"cn">>,
+                 <<"memberattr">> => <<"memberUid">>,
+                 <<"memberattr_format">> => <<"%u">>,
+                 <<"memberattr_format_re">> => <<"">>,
+                 <<"auth_check">> => true,
+                 <<"user_cache_validity">> => ?USER_CACHE_VALIDITY,
+                 <<"group_cache_validity">> => ?GROUP_CACHE_VALIDITY,
+                 <<"user_cache_size">> => ?CACHE_SIZE,
+                 <<"group_cache_size">> => ?CACHE_SIZE,
+                 <<"rfilter">> => <<"">>,
```

Comment:
Docs say `"(objectClass=inetOrgPerson)"`...

Reply:
Yeees... good catch! I think the docs were incorrect in this case.